snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedReceivedSnapshot.java

```diff
@@ -173,7 +174,17 @@ private boolean isSnapshotIdInvalid(final String snapshotId) {
 
   private boolean writeReceivedSnapshotChunk(
       final SnapshotChunk snapshotChunk, final Path snapshotFile) throws IOException {
-    Files.write(snapshotFile, snapshotChunk.getContent(), CREATE_NEW, StandardOpenOption.WRITE);
+    final var contents = ByteBuffer.wrap(snapshotChunk.getContent());
+    try (final var channel =
+        FileChannel.open(snapshotFile, StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE)) {
+      while (contents.hasRemaining()) {
+        final var limit = Math.min(IoUtil.BLOCK_SIZE, contents.remaining());
+        channel.write(contents.limit(contents.position() + limit));
+      }
+
+      channel.force(false);
+    }
+
```

Comment:
Tbh. It was totally unexpected for me to have this in this PR :sweat_smile: Maybe we can create next time a different PR. Is there an issue we want to fix with that? 

> Also reduces the number of copies necessary by removing the need for an intermediary buffer used in Files#write.

Have we seen issues with that?

If you take a look at the java doc then you see that it say it closes after writing to it, so I would assume that the content needs to be written otherwise it wouldn't make sense?

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Files.html#write(java.nio.file.Path,java.lang.Iterable,java.nio.charset.Charset,java.nio.file.OpenOption...)
> public static Path write​(Path path, Iterable<? extends CharSequence> lines, Charset cs, OpenOption... options) throws IOException
> Write lines of text to a file. Each line is a char sequence and is written to the file in sequence with each line terminated by the platform's line separator, as defined by the system property line.separator. Characters are encoded into bytes using the specified charset.
> The options parameter specifies how the file is created or opened. If no options are present then this method works as if the CREATE, TRUNCATE_EXISTING, and WRITE options are present. In other words, it opens the file for writing, creating the file if it doesn't exist, or initially truncating an existing regular-file to a size of 0. The method ensures that the file is closed when all lines have been written (or an I/O error or other runtime exception is thrown). If an I/O error occurs then it may do so after the file has been created or truncated, or after some bytes have been written to the file.

The previous code looked way simpler and I would prefer to use the java utilities instead of implementing such things on our own.